synapse/http/matrixfederationclient.py

```diff
@@ -1114,6 +1132,74 @@ async def get_json(
             Succeeds when we get a 2xx HTTP response. The
             result will be the decoded JSON body.
 
+        Raises:
+            HttpResponseException: If we get an HTTP response code >= 300
+                (except 429).
+            NotRetryingDestination: If we are not yet ready to retry this
+                server.
+            FederationDeniedError: If this destination is not on our
+                federation whitelist
+            RequestSendFailed: If there were problems connecting to the
+                remote, due to e.g. DNS failures, connection timeouts etc.
+        """
+        json_dict, _ = await self.get_json_with_headers(
+            destination=destination,
+            path=path,
+            args=args,
+            retry_on_dns_fail=retry_on_dns_fail,
+            timeout=timeout,
+            ignore_backoff=ignore_backoff,
+            try_trailing_slash_on_400=try_trailing_slash_on_400,
+            # TODO: type-ignore: can we resolve this? Argument "parser" to "get_json_with_headers" of "MatrixFederationHttpClient" has incompatible type "Optional[ByteParser[T]]"; expected "Optional[ByteParser[Union[Dict[str, Any], T]]]"
+            parser=parser,  # type: ignore[arg-type]
```

Comment:
Not sure how to resolve this correctly:

```
synapse/http/matrixfederationclient.py:1153: error: Argument "parser" to "get_json_with_headers" of "MatrixFederationHttpClient" has incompatible type "Optional[ByteParser[T]]"; expected "Optional[ByteParser[Union[Dict[str, Any], T]]]"  [arg-type]
```

The only reason I introduced `get_json_with_headers(...)` is for test purposes (see usage)

Reply:
Does the [`reveal_type(...)`](https://mypy.readthedocs.io/en/stable/common_issues.html#reveal-type) trick [mentioned by @DMRobertson](https://matrix.to/#/!vcyiEtMVHIhWXcJAfl:sw1v.org/$JpETAvKjUkSp-a4Qm8FU_kj84d3uq_cfdRPe_0KYRHU?via=matrix.org&via=element.io&via=pixie.town) help me debug this? :thinking:

Reply:
Fixed by adding the same `@overload`'s to `get_json_with_headers(...)` that `get_json(...)` has :white_check_mark: